numpy/f2py/tests/test_callback.py

```diff
@@ -211,3 +211,27 @@ class TestF77CallbackPythonTLS(TestF77Callback):
     compiler-provided
     """
     options = ["-DF2PY_USE_PYTHON_TLS"]
+
+
+class TestF90Callback(util.F2PyTest):
+
+    suffix = '.f90'
+
+    code = textwrap.dedent("""
+    function gh17797(f, y) result(r)
```

Comment:
Should have one more level of indentation for the code.

Reply:
Something like
```
    code = textwrap.dedent(
        """
        function gh17797(f, y) result(r)
          external f
          integer(8) :: r, f
          integer(8), dimension(:) :: y
          r = f(0)
          r = r + sum(y)
        end function gh17797
        """)
```

Reply:
Suggestion applied.